jsonrpc/jsonrpc.go

```diff
@@ -111,12 +113,23 @@ func (j *JSONRPC) setupHTTP() error {
 		ReadHeaderTimeout: 60 * time.Second,
 	}
 
-	if j.config.TLSCertFile != "" && j.config.TLSKeyFile != "" {
-		j.logger.Info("TLS", "cert file", j.config.TLSCertFile)
-		j.logger.Info("TLS", "key file", j.config.TLSKeyFile)
+	if j.config.UseTLS {
+		j.logger.Info("configuring http server with tls...")
+
+		cert, err := loadTLSCertificate(j.config.SecretsManager)
+		if err != nil {
+			j.logger.Error("loading tls certificate", "err", err)
+
+			return err
+		}
+
+		srv.TLSConfig = &tls.Config{
+			Certificates: []tls.Certificate{*cert},
+			MinVersion:   tls.VersionTLS12,
+		}
 
 		go func() {
-			if err := srv.ServeTLS(lis, j.config.TLSCertFile, j.config.TLSKeyFile); err != nil {
+			if err := srv.ServeTLS(lis, "", ""); err != nil {
```

Comment:
Just to confirm if my understanding is correct:
in case we manually privde the `tls.Config` to the `http.Server` instance, we can omit the `certFile` and `keyFile` from this function?

Reply:
Cert and key file get now from secrets manager, as it is for other keys (validator and libp2p). Hence there is no need for defining path to those files, they will be placed at the same location where other keys are.